### ISSUE TITLE
Add support for AWS directconnect.

### ISSUE THREAD

**Issue by robh007** [infracost/infracost#414]
I think this is the last of the connectivity resources that are missing, now data transfer is able to be calculated it maybe worth adding direct connect.

```
aws_dx_bgp_peer
aws_dx_connection
aws_dx_connection_association
aws_dx_gateway
aws_dx_gateway_association
aws_dx_gateway_association_proposal
aws_dx_hosted_private_virtual_interface
aws_dx_hosted_private_virtual_interface_accepter
aws_dx_hosted_public_virtual_interface
aws_dx_hosted_public_virtual_interface_accepter
aws_dx_hosted_transit_virtual_interface
aws_dx_hosted_transit_virtual_interface_accepter
aws_dx_lag
aws_dx_private_virtual_interface
aws_dx_public_virtual_interface
aws_dx_transit_virtual_interface
```

**Comment by robh007** [comment]
I'll pick this one up @alikhajeh1

**Comment by alikhajeh1** [comment]
Released in v0.8.0
